internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1593,7 +1594,13 @@ func ResetSystemDisk(seq runtime.Sequence, data interface{}) (runtime.TaskExecut
 	return func(ctx context.Context, logger *log.Logger, r runtime.Runtime) (err error) {
 		var dev *blockdevice.BlockDevice
 
-		dev, err = blockdevice.Open(r.State().Machine().Disk().Device().Name())
+		disk := r.State().Machine().Disk(disk.WithPartitionLabel(constants.BootPartitionLabel))
```

Comment:
Changed reset to lookup the system disk by the `BootPartitionLabel`. It feels like this is a more appropriate way to do it.
As another option we can do lookup by ephemeral first then fall back to boot.

Reply:
I'm not sure what the consequences would be, but my guess we should always reset by EPHEMERAL, as this is the most critical one used everywhere. Imagine having an USB stick plugged in with no EPHEMERAL in it, but BOOT.

In maintenance mode, we should figure out if the Talos is installed, and if we do, we know the system disk.